src/main/resources/org/opentripplanner/apis/gtfs/schema.graphqls

```diff
@@ -1253,6 +1510,213 @@ input ParkingFilter {
+    should be `FeedId:StopLocationId`.
+    """
+    stopLocationId: String!
+}
+
+"""
+Mode selections for the plan search.
+"""
+input PlanModesInput {
+    """
+    Should only the direct search without any transit be done.
+    """
+    directOnly: Boolean = false
+
+    """
+    Should only the transit search be done and never suggest itineraries that don't
+    contain any transit legs.
+    """
+    transitOnly: Boolean = false
+
+    """
+    Street mode that is used when searching for itineraries that don't use any transit.
+    If more than one mode is selected, at least one of them must be used but not necessarily all.
+    There are modes that automatically also use walking such as the rental modes. To force rental
+    to be used, this should only include the rental mode and not `WALK` in addition.
+    The default access mode is `WALK`.
+    """
+    direct: [PlanDirectMode!]
+
+    """
+    Modes for different phases of an itinerary when transit is included. Also
+    includes street mode selections related to connecting to the transit network
+    and transfers. By default, all transit modes are usable and `WALK` is used for
+    access, egress and transfers.
+    """
+    transit: PlanTransitModesInput
+}
+
+"""
+Modes for different phases of an itinerary when transit is included. Also includes street
+mode selections related to connecting to the transit network and transfers.
+"""
+input PlanTransitModesInput {
+    """
+    Street mode that is used when searching for access to the transit network from origin.
+    If more than one mode is selected, at least one of them must be used but not necessarily all.
+    There are modes that automatically also use walking such as the rental modes. To force rental
+    to be used, this should only include the rental mode and not `WALK` in addition.
+    The default access mode is `WALK`.
+    """
+    access: [PlanAccessMode!]
+
+    """
+    Street mode that is used when searching for egress to destination from the transit network.
+    If more than one mode is selected, at least one of them must be used but not necessarily all.
+    There are modes that automatically also use walking such as the rental modes. To force rental
+    to be used, this should only include the rental mode and not `WALK` in addition.
+    The default access mode is `WALK`.
+    """
+    egress: [PlanEgressMode!]
+
+    """
+    Street mode that is used when searching for transfers. Selection of only one allowed for now.
+    The default transfer mode is `WALK`.
+    """
+    transfer: [PlanTransferMode!]
+
+    """
+    Transit modes and reluctances associated with them. Each defined mode can be used in
+    an itinerary but doesn't have to be. If direct search is not disabled, there can be an
+    itinerary without any transit legs. By default, all transit modes are usable.
+    """
+    transit: [PlanTransitModePreferenceInput!]
```

Comment:
Btw what should an empty list mean? Right now it's just ignored and default modes are used. We have explicit field for disabling transit search. If the list is empty, we could:
1. Throw an exception
2. Disable transit routing
3. Use default mode
This isn't unique to this field. We should make some decision on how empty lists should be interpreted in general. I think I might have ignored the input in all cases but maybe I shouldn't have. For example, when banning some things there isn't other way to override default banning.

Reply:
Until we have worked out what it means we should throw an exception so that whatever expectation that people have doesn't get cemented.

Can you remind me why we went with an explicit field for enabling/disabling transit?

Reply:
@t2gran came up with this design but I think the reason is that it's more explicit.